packages/access-client/src/agent.js

```diff
@@ -28,6 +32,59 @@ export { AgentData }
 const HOST = 'https://access.web3.storage'
 const PRINCIPAL = DID.parse('did:web:web3.storage')
 
+/**
+ *
+ * @param {string} email
+ * @returns {Ucanto.Principal<Ucanto.DID<'mailto'>>}
+ */
+function emailToSessionPrincipal(email) {
+  const parts = email.split('@').map((s) => encodeURIComponent(s))
+  return DID.parse(`did:mailto:${parts[1]}:${parts[0]}`)
```

Comment:
Might be a good idea to verify that you have two parts and non are empty. I remember seeing cases where that was not the case.

Reply:
idea: we should have a `@web3-storage/did-mailto` package in this monorepo with e.g. canonical type for this and maybe also a ucanto validator schema that matches the type.

Reply:
https://github.com/web3-storage/w3protocol/pull/433/commits/0cc4e47fd8383927393c32e7d06fc5e56f0fe10a